src/Cask/Helpers.cs

```diff
@@ -70,36 +69,32 @@ public static byte KindToByte(CaskKeyKind kind)
         return (byte)((int)kind << CaskKindReservedBits);
     }
 
-    public static byte ProviderKindToByte(string providerKind)
+    public static byte ProviderKindToByte(char providerKind)
     {
-        Debug.Assert(providerKind?.Length == 1, "Provider kind should be a single character.");
-
         int base64Index;
 
         const int uppercaseZIndex = 25; // 'Z' - 'A';
         const int lowercaseZIndex = 51; // 'z' - 'a';
```

Comment:
```suggestion
        const int uppercaseZIndex = 'Z' - 'A';
        const int lowercaseZIndex = 'z' - 'a';
```
From the prior PR, but there's no benefit to hard-coding + comment here. The compiler can do this for you. The generated code will be identical so no runtime cost.

Reply:
Thank you compiler.